### PR TITLE
map

### PR DESCRIPTION
**Pull request by shashi**

### CODE REVIEW

**Comment by shashi** [comment]
Would be great if this can be cherry-picked onto `v2.0.0` and a `v2.1.0` is made available so that we can use this in IndexedTables.jl (https://github.com/JuliaLang/METADATA.jl/pull/8476#issuecomment-288996616)

**Comment by mdcfrancis** [comment]
Looks good let me have a look at getting this on v2.1.0

**Comment by shashi** [comment]
Hold on, I might have a small optimization. Will update in a bit. Thanks!

**Comment by shashi** [comment]
updated

File: src/NamedTuples.jl [hunk 1]
```diff
@@ -258,6 +258,28 @@ else
...
+    for x in nts[2:end]
+        if !isequal(fieldnames(x), fields)
+            throw(ArgumentError("All NamedTuple inputs to map must have the same fields in the same order"))
+        end
+    end
+    N = nfields(nts[1])
+    M = length(nts)
+
+    NT = create_tuple(fields) # This type will already exist if this function may be called
+    args = Expr[:(f($(Expr[:(nts[$i][$j]) for i = 1:M]...))) for j = 1:N]
```
Comment (JeffBezanson) [inline src/NamedTuples.jl]: This should use `:(getfield(nts[$i],$j))`

**Comment by shashi** [comment]
I've applied the patch over v2.0.0 resolving conflicts. https://github.com/shashi/NamedTuples.jl/tree/map-backport - this branch should be tagged as v2.1.0

It would also help to keep a `release-0.5` branch in this repo so that PRs can be made against it (I couldn't open a PR against a tag)

**Comment by davidanthoff** [comment]
> It would also help to keep a release-0.5 branch in this repo so that PRs can be made against it (I couldn't open a PR against a tag)

Ideally that branch would be named release-2.x, i.e. refer to the NamedTuples version that is being maintained, not the julia version.

Anything else holding things up here? I would really like to see IndexedTables tagged and registered, and this seems to be the hold up right now?

**Comment by mdcfrancis** [comment]
I added a release-julia-0.5 branch, but can rename to the NamedTuples version?

**Comment by shashi** [comment]
I think _some_ branch will do, the name is immaterial. `release-julia-0.5` makes more sense than `release-0.5` for sure.

**Comment by davidanthoff** [comment]
I agree, I care more about a speedy merge and tag of all this than the name :)

**Comment by mdcfrancis** [comment]
working on it now

**Pull request merged**